sdks/java/core/src/test/java/org/apache/beam/sdk/io/AvroIOTest.java

```diff
@@ -1230,21 +1349,43 @@ public void testDynamicDestinationsWithNumShards() throws Exception {
     @Category(NeedsRunner.class)
     public void testDynamicDestinationsViaSinkRunnerDeterminedSharding() throws Exception {
       testDynamicDestinationsUnwindowedWithSharding(
-          WriteMethod.AVROIO_SINK, Sharding.RUNNER_DETERMINED);
+          WriteMethod.AVROIO_SINK_WITH_SCHEMA, Sharding.RUNNER_DETERMINED);
```

Comment:
Why `WriteMethod.AVROIO_SINK` has been changed to `WriteMethod.AVROIO_SINK_WITH_SCHEMA` and not to `WriteMethod.AVROIO_SINK_WITH_CLASS` ?

Reply:
All of the tests going to `testDynamicDestinationsUnwindowedWithSharding` exercise the non-specific ("GenericRecord") implementation.  There is no specific class ( or `Class<...>`) associated with the schemas in this test.  

In the past `WriteMethod` flipped between `write()` and `sink`, now it distinguishes between the different sink methods explicitly (hopefully making the test more readable!).